src/dmd/dsymbolsem.d

```diff
@@ -5989,3 +5986,48 @@ void aliasSemantic(AliasDeclaration ds, Scope* sc)
             ScopeDsymbol.multiplyDefined(Loc.initial, sx, ds);
     }
 }
+
+/**
+ * build a shim function around the compound dtor that translates
+ *  a C++ destructor to a destructor with extern(D) calling convention
+ *
+ * Params:
+ *  cldec = the class that contains the destructor to wrap
+ *  sc = the scope in which to analyze the new function
+ *
+ * Returns:
+ *  the shim destructor, semantically analyzed and added to the class as a member
+ */
+private DtorDeclaration buildExternDDtor(ClassDeclaration cldec, Scope* sc)
```

Comment:
perhaps this function should go into clone.d along with the other Dtor building functions.

Reply:
Ok, will move.